multi-node/aws/pkg/config/config.go

```diff
@@ -385,26 +426,78 @@ func (cfg Cluster) valid() error {
 		return fmt.Errorf("invalid vpcCIDR: %v", err)
 	}
 
-	_, instancesNet, err := net.ParseCIDR(cfg.InstanceCIDR)
-	if err != nil {
-		return fmt.Errorf("invalid instanceCIDR: %v", err)
-	}
-	if !vpcNet.Contains(instancesNet.IP) {
-		return fmt.Errorf("vpcCIDR (%s) does not contain instanceCIDR (%s)",
-			cfg.VPCCIDR,
-			cfg.InstanceCIDR,
-		)
-	}
-
 	controllerIPAddr := net.ParseIP(cfg.ControllerIP)
 	if controllerIPAddr == nil {
 		return fmt.Errorf("invalid controllerIP: %s", cfg.ControllerIP)
 	}
-	if !instancesNet.Contains(controllerIPAddr) {
-		return fmt.Errorf("instanceCIDR (%s) does not contain controllerIP (%s)",
-			cfg.InstanceCIDR,
-			cfg.ControllerIP,
-		)
+
+	if len(cfg.Subnets) == 0 {
```

Comment:
Question: If we move [this](https://github.com/mumoshu/coreos-kubernetes/blob/multi-az-workers/multi-node/aws/pkg/config/config.go#L87) block of logic __before__ `c.valid()`, can't we do away with this conditional validation behavior and only validate the `cfg.Subnets` array, as the backwards compatibility measure would have already been taken?